okio/src/main/java/okio/BufferedSource.java

```diff
@@ -381,6 +381,28 @@
   String readUtf8Line() throws IOException;
 
   /**
+   * Like {@link #readUtf8Line()}, but allows the caller to specify the longest
+   * allowed match.
+   *
+   * <p>The returned string will have at most {@code limit} UTF-8 bytes, and the maximum number
+   * of bytes scanned is {@code limit + 2}. If {@code limit == 0} this will return an empty string.
```

Comment:
I wonder if it’s simpler to be explicit about which bytes are behind the `+2` ? Like _“consumes up to limit bytes plus `\n` or `\r\n`”_